db.go

```diff
@@ -940,11 +940,31 @@ func (db *DB) Snapshot(dir string, withHead bool) error {
 			return errors.Wrapf(err, "error snapshotting block: %s", b.Dir())
 		}
 	}
+
 	if !withHead {
 		return nil
 	}
-	_, err := db.compactor.Write(dir, db.head, db.head.MinTime(), db.head.MaxTime(), nil)
-	return errors.Wrap(err, "snapshot head block")
+
+	mint := db.head.MinTime()
+	maxt := db.head.MaxTime()
+
+	// Wrap head into a range that bounds all reads to it.
+	// Otherwise we can have race between appends and head.Chunks() that takes all it has.
```

Comment:
This doesn't eliminate that race, as a single chunk can be extended beyond the max.

Reply:
Yes  yes - it's leftover comment, sorry. I found it just more explicit to wrap it with range - it will at least not fetch chunks that starts after maxTime of the moment of invoking `Write`. The actual race is solved on `populateBlock` with injecting extra `drange`